variant_to_realignment.nf

```diff
@@ -0,0 +1,274 @@
+        path "variants_read2.fa", emit: variants_read2
+
+    '''
+    # Get the fasta sequences for these intervals
+    bedtools getfasta -fi genome.fa -bed flanking_r1.bed -fo variants_read1.fa
+    bedtools getfasta -fi genome.fa -bed flanking_r2.bed -fo variants_read2.fa
+
+    # Replace the colon separators with "|":
+    # Storing this information for later on in the script when we split the name by "|" to extract the relevant 
+    # information (ALT allele, QUAL, FILT, INFO)
+    # This is done because the INFO column can contain ":", which means we wouldn't be able to split by ":", so "|" was 
+    # chosen
+    sed -i 's/:/|/' variants_read1.fa
+    sed -i 's/:/|/' variants_read2.fa
+    '''
+}
+
+/*
+ * Extract information about the original variants and put it in the fasta header
+ */
+process extractVariantInfoToFastaHeader {
+
+    input:  
+        path "flanking_r1.bed"
+        path "flanking_r2.bed"
+        path "variants_read1.fa"
+        path "variants_read2.fa"
+
+    output:
+        path "variant_read1.out.fa", emit: variant_read1_with_info
+        path "variant_read2.out.fa", emit: variant_read2_with_info
+
+    // Disable the string interpolation using single quotes
+    // https://www.nextflow.io/docs/latest/script.html#string-interpolation
+    '''
+    # Store variant positions
+    cut -f 1,3 flanking_r1.bed > position.txt
+
+    # Store ref bases
+    cut -f 6 flanking_r1.bed > old_ref_bases.txt
+
+    # Store rsIDs
+    cut -f 4 flanking_r1.bed > rsIDs.txt
+
+    # Store variant bases
+    cut -f 7  flanking_r1.bed > variant_bases.txt
+
+    # Store the other vcf columns
+    cut -f 5,8,9 flanking_r1.bed > qual_filt_info.txt
+
+    # Paste the names, variant bases, then fasta sequences into a new file
+    paste position.txt old_ref_bases.txt variant_bases.txt rsIDs.txt qual_filt_info.txt \
+    <(grep -v '^>' variants_read1.fa) | awk '{print ">"$0}' > temp1.txt
+
+    paste position.txt old_ref_bases.txt variant_bases.txt rsIDs.txt qual_filt_info.txt \
+    <(grep -v '^>' variants_read2.fa) | awk '{print ">"\$0}' > temp2.txt
+
+    # Reformat the fasta ID: inconsistencies in the separators, and no new line before the sequence mean that this next 
+    # command is a bit ugly
+    # Input:
+    # >[chr]|[pos interval]   [REF]       [ALT]       [rsID]    [QUAL] [FILT] [INFO]      [seq]
+    # Replace all spaces and tabs with "|":
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]|[seq]
+    # Replace the last "|" with a space (this is between the header and the sequence):
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO] [seq]
+    # And finally replace the space with a newline:
+    # Output:
+    # >[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]
+    # [seq]
+    sed 's/\\t/|/g; s/ /|/g; s/\\(.*\\)|/\\1 /' temp1.txt | tr ' ' '\\n' > variant_read1.out.fa
+    sed 's/\\t/|/g; s/ /|/g; s/\\(.*\\)|/\\1 /' temp2.txt | tr ' ' '\\n' > variant_read2.out.fa
```

Comment:
I just realised something: the flanking region “reads” that we generate are the same orientation. In other words, we expect them to map in either FF or RR orientation. This kind of goes against the assumption that the paired-end sequencing reads are expected to map FR or RF. It seems to be working fine for a moment, but I wonder if it might trip some aligners under some circumstances.

(I don't think it's worth changing this at this point, just an observation.)

Reply:
Yes that is correct. And I don't think this is a problem at all. Most aligners don't make assumption about orientation anymore only about distances. I know Minimap2 does not seem to care about the orientation